redis/src/main/scala/zio/redis/options/Connection.scala

```diff
@@ -27,38 +33,98 @@ trait Connection {
+      case 'O' => MonitorMode
+      case 'P' => PubSub
+      case 'r' => ReadOnlyMode
+      case 'S' => Replica
+      case 'u' => Unblocked
+      case 'U' => UnixDomainSocket
+      case 'x' => MultiExecContext
+      case 't' => KeysTrackingEnabled
+      case 'R' => TrackingTargetClientInvalid
+      case 'B' => BroadcastTrackingMode
+      case 'N' => NoSpecificFlagSet
+      case x   => UnknownFlag(x)
+    }
 
-    case object Blocked extends ClientFlag
+    case object ToBeClosedAsap extends ClientFlag {
+      private[redis] def stringify: String = "A"
+    }
 
-    case object ToBeClosedAfterReply extends ClientFlag
+    case object Blocked extends ClientFlag {
+      private[redis] def stringify: String = "b"
+    }
 
-    case object WatchedKeysModified extends ClientFlag
+    case object ToBeClosedAfterReply extends ClientFlag {
+      private[redis] def stringify: String = "c"
+    }
 
-    case object IsMaster extends ClientFlag
+    case object WatchedKeysModified extends ClientFlag {
+      private[redis] def stringify: String = "d"
+    }
 
-    case object MonitorMode extends ClientFlag
+    case object IsMaster extends ClientFlag {
+      private[redis] def stringify: String = "M"
+    }
 
-    case object PubSub extends ClientFlag
+    case object NoSpecificFlagSet extends ClientFlag {
+      private[redis] def stringify: String = "N"
+    }
 
-    case object ReadOnlyMode extends ClientFlag
+    case object MonitorMode extends ClientFlag {
+      private[redis] def stringify: String = "O"
+    }
 
-    case object Replica extends ClientFlag
+    case object PubSub extends ClientFlag {
+      private[redis] def stringify: String = "P"
+    }
 
-    case object Unblocked extends ClientFlag
+    case object ReadOnlyMode extends ClientFlag {
+      private[redis] def stringify: String = "r"
+    }
 
-    case object UnixDomainSocket extends ClientFlag
+    case object Replica extends ClientFlag {
+      private[redis] def stringify: String = "S"
+    }
 
-    case object MultiExecContext extends ClientFlag
+    case object Unblocked extends ClientFlag {
+      private[redis] def stringify: String = "u"
+    }
 
-    case object KeysTrackingEnabled extends ClientFlag
+    case object UnixDomainSocket extends ClientFlag {
+      private[redis] def stringify: String = "U"
+    }
 
-    case object TrackingTargetClientInvalid extends ClientFlag
+    case object MultiExecContext extends ClientFlag {
+      private[redis] def stringify: String = "x"
+    }
 
-    case object BroadcastTrackingMode extends ClientFlag
+    case object KeysTrackingEnabled extends ClientFlag {
+      private[redis] def stringify: String = "t"
+    }
+
+    case object TrackingTargetClientInvalid extends ClientFlag {
+      private[redis] def stringify: String = "R"
+    }
+
+    case object BroadcastTrackingMode extends ClientFlag {
+      private[redis] def stringify: String = "B"
+    }
+
+    case class UnknownFlag(flag: Char) extends ClientFlag {
```

Comment:
Let's be consistent and either make it final or sealed.

Reply:
updated this in the last commit (merged with master)